src/certifier.cc

```diff
@@ -246,6 +246,101 @@ bool GetX509FromCert(const string& cert, X509* x) {
   return asn1_to_x509(cert, x);
 }
 
+#ifdef SEV_SNP
+
+#define EXT_STRUCT_VERSION  "1.3.6.1.4.1.3704.1.1"
+#define EXT_PRODUCT_NAME    "1.3.6.1.4.1.3704.1.2"
+#define EXT_BLSPL           "1.3.6.1.4.1.3704.1.3.1"
+#define EXT_TEESPL          "1.3.6.1.4.1.3704.1.3.2"
+#define EXT_SNPSPL          "1.3.6.1.4.1.3704.1.3.3"
+#define EXT_SPL4            "1.3.6.1.4.1.3704.1.3.4"
+#define EXT_SPL5            "1.3.6.1.4.1.3704.1.3.5"
+#define EXT_SPL6            "1.3.6.1.4.1.3704.1.3.6"
+#define EXT_SPL7            "1.3.6.1.4.1.3704.1.3.7"
+#define EXT_UCODESPL        "1.3.6.1.4.1.3704.1.3.8"
+#define EXT_HWID            "1.3.6.1.4.1.3704.1.4"
+
+static bool vcek_ext_byte_value(X509 *vcek, const char *oid, unsigned char *value) {
+  int nid = -1, idx = -1, extlen = -1;
+  X509_EXTENSION *ex = NULL;
+  ASN1_STRING *extvalue = NULL;
+  unsigned char *vals = NULL;
+
+  // Use OID for both lname and sname so OBJ_create does not fail
+  nid = OBJ_create(oid, oid, oid);
+  if (nid == NID_undef) {
+    printf("Failed to create NID\n");
+    return false;
+  }
+  idx = X509_get_ext_by_NID(vcek, nid, -1);
+  if (idx == -1) {
+    return false;
+  }
+
+  ex = X509_get_ext(vcek, idx);
+  extvalue = X509_EXTENSION_get_data(ex);
+  extlen = ASN1_STRING_length(extvalue);
+  vals = (unsigned char *) ASN1_STRING_get0_data(extvalue);
```

Comment:
Is there a reason we are removing the `const` context here from openssl? It doesn't look like we are modifying it in any way. If we are going to cast, would it make more sense to do:

```suggestion
  vals = reinterpret_cast<unsigned char *>(ASN1_STRING_get0_data(extvalue));
```